internal/driver/deviceclient.go

```diff
@@ -105,126 +105,186 @@ func calculateAddressLength(primaryTable string, valueType sdkModel.ValueType) u
 	return length
 }
 
-func TransformDataBytesToResult(req *sdkModel.CommandRequest, dataBytes []byte, commandInfo *CommandInfo) (*sdkModel.CommandValue, error) {
-	var result = &sdkModel.CommandValue{}
+func TransformDataBytesToResult(req *models.CommandRequest, dataBytes []byte, commandInfo *CommandInfo) (*models.CommandValue, error) {
 	var err error
+	var result = &models.CommandValue{}
 	var resTime = time.Now().UnixNano()
 
 	switch commandInfo.ValueType {
-	case sdkModel.Uint16:
+	case v2.ValueTypeUint16:
 		var res = binary.BigEndian.Uint16(dataBytes)
-		result, err = sdkModel.NewUint16Value(req.DeviceResourceName, resTime, res)
-	case sdkModel.Uint32:
+		result, err = models.NewCommandValue(req.DeviceResourceName, v2.ValueTypeUint16, res)
+		if err != nil {
+			return nil, err
+		}
+		result.Origin = resTime
```

Comment:
looks like this block of repeated code can now go after the switch if `res` is define before the stitch as: `var res interface{}`. may sure res is set/cast appropriately in switch code to allow this.